app/controllers/users_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def shopping_list
+    @shopping_list = []
+    all_foods = Food.all
+    foods_in_recipes = Food.joins(:recipes).distinct
+    all_foods.each do |food|
+      @shopping_list << food unless foods_in_recipes.include?(food)
+    end
+    @total_item = @shopping_list.count
+    @total_price = @shopping_list.sum { |food| food.price * food.quantity }
+  end
```

Comment:
- [ ] Please show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user) but for now I didn't see any data on the shopping list it's totally empty tables if the user add Ingredient food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user) so kindly fix this issue as well 
- [ ] Please count the total food items and the total price of the missing food and show it in the shopping list.

Reply:
@aamirkhan2478 can you explain us more the general shopping list